tools/CMakeLists.txt

```diff
@@ -15,323 +15,381 @@ endif()
 # to be filled with all targets the tools subsystem
 set(PCL_TOOLS_ALL_TARGETS)
 
-PCL_ADD_EXECUTABLE(pcl_sac_segmentation_plane COMPONENT ${SUBSYS_NAME} SOURCES sac_segmentation_plane.cpp)
-target_link_libraries(pcl_sac_segmentation_plane pcl_common pcl_io pcl_sample_consensus pcl_segmentation)
+PCL_ADD_EXECUTABLE(pcl_pcd_convert_NaN_nan COMPONENT ${SUBSYS_NAME} SOURCES pcd_convert_NaN_nan.cpp)
 
-PCL_ADD_EXECUTABLE(pcl_plane_projection COMPONENT ${SUBSYS_NAME} SOURCES plane_projection.cpp)
-target_link_libraries (pcl_plane_projection pcl_common pcl_io pcl_sample_consensus)
+if(TARGET pcl_sample_consensus)
+  PCL_ADD_EXECUTABLE(pcl_plane_projection COMPONENT ${SUBSYS_NAME} SOURCES plane_projection.cpp)
+  target_link_libraries (pcl_plane_projection pcl_common pcl_io pcl_sample_consensus)
+endif()
 
-PCL_ADD_EXECUTABLE(pcl_normal_estimation COMPONENT ${SUBSYS_NAME} SOURCES normal_estimation.cpp)
-target_link_libraries (pcl_normal_estimation pcl_common pcl_io pcl_features pcl_kdtree)
+if(TARGET pcl_io)
```

Comment:
Check for `TARGET pcl_io` can be omitted because io is a "hard" dependency (2 more checks like this)